cmd/ci-operator-configresolver/main.go

```diff
@@ -0,0 +1,189 @@
+package main
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"net/http"
+	"os"
+	"path/filepath"
+
+	"github.com/fsnotify/fsnotify"
+	log "github.com/sirupsen/logrus"
+
+	"github.com/openshift/ci-tools/pkg/api"
+	"github.com/openshift/ci-tools/pkg/config"
+	"github.com/openshift/ci-tools/pkg/load"
+	"github.com/openshift/ci-tools/pkg/registry"
+	types "github.com/openshift/ci-tools/pkg/steps/types"
+)
+
+type options struct {
+	resolver   registry.Resolver
+	configPath string
+	regPath    string
+}
+
+func resolveConfig(o *options) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		if r.Method != "GET" {
+			w.WriteHeader(http.StatusNotImplemented)
+			w.Write([]byte(http.StatusText(http.StatusNotImplemented)))
+			return
+		}
+		org := r.URL.Query().Get("org")
+		if org == "" {
+			w.WriteHeader(http.StatusBadRequest)
+			w.Write([]byte("org query missing or incorrect"))
+			return
+		}
+		repo := r.URL.Query().Get("repo")
+		if repo == "" {
+			w.WriteHeader(http.StatusBadRequest)
+			w.Write([]byte("repo query missing or incorrect"))
+			return
+		}
+		branch := r.URL.Query().Get("branch")
+		if branch == "" {
+			w.WriteHeader(http.StatusBadRequest)
+			w.Write([]byte("branch query missing or incorrect"))
+			return
+		}
+		variant := r.URL.Query().Get("variant")
+		info := config.Info{
+			Org:     org,
+			Repo:    repo,
+			Branch:  branch,
+			Variant: variant,
+		}
+		info.Filename = filepath.Join(o.configPath, branch, info.Basename())
+
+		config, err := load.Config(info.Filename)
```

Comment:
I think I was expecting to see a server that has an up-to-date mapping `config.ByFilename`, validate on load (which happens on file changes) and not load or validate on user request

Reply:
Ok, I can do that